ci/upload_github_release_assets.go

```diff
@@ -157,38 +170,65 @@ func mustUpdateFormulas(ctx context.Context, versionBeingReleased *versionutils.
 	}
 }
 
-func validateReleaseVersionOfCli() {
-	releaseVersion := getReleaseVersionOrExitGracefully().String()[1:]
+func validateReleaseVersionOfCli(dryRun bool) {
+	releaseVersion := getReleaseVersionOrExitGracefully(dryRun).String()[1:]
```

Comment:
nit: Could we pass a `versionutils.Version` variable into `validateReleaseVersionOfCli` instead of re-computing the version on line 174? I think then we could just pass `versionBeingReleased` into `validateReleaseVersionOfCli` when we call it on line 45

The suggestion is pseudocode, so YMMV, but something like
```suggestion
func validateReleaseVersionOfCli(version *versionutils.Version) {
	releaseVersion := version.String()[1:]
```

If we also just want to push this through to unblock CI, I'm more than happy to handle these nits in a separate PR

Reply:
let's handle this in another PR. Resolving the thread